client.go

```diff
@@ -23,6 +24,17 @@ func (b *Backend) clientMongo(ctx context.Context, s logical.Storage) (*mongodba
 	if err != nil {
 		return nil, err
 	}
+
+	pluginEnv, err := b.system.PluginEnv(ctx)
+	if err != nil {
+		return nil, err
+	}
+	if pluginEnv != nil {
+		client.UserAgent = useragent.PluginString(pluginEnv, "mongodbatlas-secrets")
+	} else {
+		client.UserAgent = useragent.String()
+	}
```

Comment:
nit: we can make this a little more compact

```suggestion
        client.UserAgent = useragent.String()
	if pluginEnv != nil {
		client.UserAgent = useragent.PluginString(pluginEnv, "mongodbatlas-secrets")
	}
```

Reply:
Thanks for the suggestion, imho it flows a little better for readability as-is, so I'm going to leave it in the absence of any strong opinions.